src/index.ts

```diff
@@ -0,0 +1,233 @@
+import p from 'path'
+import fs from 'fs'
+import type babelCore from '@babel/core'
+import type {Primitive} from 'type-fest'
+import {getReplacement, requireFromString} from './helpers'
+
+type VisitorState = {
+  file: {
+    opts: babelCore.TransformOptions
+  }
+}
+
+export default function prevalPlugin(
+  babel: typeof babelCore,
+): babelCore.PluginObj<VisitorState> {
+  const {types: t, template, transformFromAstSync} = babel
+  const assignmentBuilder = template('const NAME = VALUE')
+  return {
+    name: 'preval',
+    visitor: {
+      Program(path, {file: {opts: fileOpts}}) {
+        const firstNode = path.node.body[0] || {}
+        const comments = firstNode.leadingComments ?? []
+        const prevalComment = comments.find(isPrevalComment)
+
+        if (!prevalComment) {
+          return
+        }
+
+        prevalComment.value = ' this file was prevaled'
+
+        const result = transformFromAstSync(path.node, undefined, fileOpts)
+
+        // istanbul ignore next because this should never happen, but TypeScript needs me to handle it
+        const string = result?.code ?? ''
+
+        const replacement = getReplacement({string, fileOpts, babel})
+
+        // istanbul ignore next because this should never happen, but TypeScript needs me to handle it
+        if (!replacement) return
```

Comment:
Confused with the reference to 'istanbul' ? What does it mean in this context?

Reply:
Istanbul is the code coverage tool used by jest (well technically NYC, but there's some sort of lineage between them).

These types of comments have been [taken from `babel-plugin-codegen`](https://github.com/kentcdodds/babel-plugin-codegen/blob/main/src/replace.ts#L24) for the same reasons. Generally we need to narrow the type to make TS happy, but we can't write a test for it so we tell the coverage checker to ignore this bit.